ocaml/xapi/xapi_observer.ml

```diff
@@ -59,9 +59,19 @@ let assert_valid_endpoints endpoints =
     )
     endpoints
 
+let assert_valid_attributes attributes =
+  List.iter
+    (fun (k, v) ->
+      if not (Tracing.validate_attribute (k, v)) then
+        failwith
+          (Printf.sprintf "The attribute: %s:%s is not a valid attribute" k v)
```

Comment:
```suggestion
        let kv = Printf.sprintf "%s:%s" k v in
        raise Api_errors.(Server_error (invalid_value, ["attributes"; kv]))
```

Reply:
Use something similar for the other errors above, to raise the proper API error.

Reply:
Done for all validators